e2e/app/page/data-page.ts

```diff
@@ -50,30 +56,91 @@ export default class DataPage extends AuthenticatedPage {
   }
 
+  // Click Add Datasets button.
+  async clickAddDatasetButton(waitForDatasetBuildPage: boolean = true): Promise<DatasetBuildPage> {
+    const addDatasetButton = await this.getAddDatasetButton();
+    await addDatasetButton.clickAndWait();
+
+    const datasetPage = new DatasetBuildPage(this.page);
+    // wait for Dataset Build page load and ready.
+    if (waitForDatasetBuildPage) {
+      await datasetPage.waitForLoad();
+    }
+    return datasetPage;
+  }
+
   /**
    * Delete cohort by look up its name using Ellipsis menu.
    * @param {string} cohortName
    */
   async deleteCohort(cohortName: string): Promise<string> {
     const cohortCard = await DataResourceCard.findCard(this.page, cohortName);
-    const menu = await cohortCard.getEllipsis();
+    const menu = cohortCard.getEllipsis();
     await menu.clickAction(EllipsisMenuAction.DELETE, false);
     const dialogContent = await (new CohortBuildPage(this.page)).deleteConfirmationDialog();
     console.log(`Cohort "${cohortName}" deleted.`);
     return dialogContent;
   }
 
+  /**
+   * Delete Dataset thru Ellipsis menu located inside the Dataset Resource card.
+   * @param {string} datasetName
+   */
+  async deleteDataset(datasetName: string): Promise<string> {
+    const dataSetCard = await DataResourceCard.findCard(this.page, datasetName);
+    const menu = dataSetCard.getEllipsis();
+    await menu.clickAction(EllipsisMenuAction.DELETE, false);
+
+    const dialog = new Dialog(this.page);
+    const dialogContentText = await dialog.getContent();
+    const deleteButton = await Button.findByName(this.page, {type: ElementType.Button, normalizeSpace: 'Delete Dataset'}, dialog);
+    await Promise.all([
+      deleteButton.click(),
+      dialog.waitUntilDialogIsClosed(),
+    ]);
+    await waitWhileLoading(this.page);
+
+    console.log(`Dataset "${datasetName}" deleted.`);
+    return dialogContentText;
+  }
+
+  /**
+   * Rename Dataset thru the Ellipsis menu located inside the Dataset Resource card.
+   * @param {string} datasetName
+   * @param {string} newDatasetName
+   */
+  async renameDataset(datasetName: string, newDatasetName: string): Promise<void> {
+    const card = await DataResourceCard.findCard(this.page, datasetName);
+    const menu = card.getEllipsis();
+    await menu.clickAction(EllipsisMenuAction.RenameDataset, false);
+
+    const dialog = new Dialog(this.page);
+
+    const newNameTextbox = new Textbox(this.page, '//*[@id="new-name"]');
+    await newNameTextbox.type(newDatasetName);
+
+    const descriptionTextarea = await Textarea.findByName(this.page, {containsText: 'Description:'}, dialog);
+    await descriptionTextarea.type('Puppeteer automation rename dataset.');
+
+    const renameButton = await Button.findByName(this.page, {normalizeSpace: 'Rename Dataset'}, dialog);
+    await Promise.all([
+      renameButton.click(),
+      dialog.waitUntilDialogIsClosed(120000), // Rename Dataset can be very slow sometimes.
```

Comment:
That doesn't make sense - I suspect something else was going on when you observed this. Is this reproducible?